File: src/main/java/com/github/simiacryptus/aicoder/openai/translate/TranslationRequestTemplate.java
package com.github.simiacryptus.aicoder.openai.translate;

import com.github.simiacryptus.aicoder.config.AppSettingsState;

import java.util.function.Function;

public enum TranslationRequestTemplate {
    XML(TranslationRequest_XML::new);
    private final Function<AppSettingsState, TranslationRequest> fn;

    TranslationRequestTemplate(Function<AppSettingsState, TranslationRequest> fn) {
        this.fn = fn;
    }

    public TranslationRequest get(AppSettingsState config) {
        return fn.apply(config);
    }
}

File: src/main/kotlin/com/github/simiacryptus/aicoder/openai/translate/TranslationRequestTemplate.kt
package com.github.simiacryptus.aicoder.openai.translate

import com.github.simiacryptus.aicoder.config.AppSettingsState

enum class TranslationRequestTemplate(private val fn: java.util.function.Function<AppSettingsState, TranslationRequest>) {
    XML({ settings: AppSettingsState? ->
        TranslationRequest_XML(
            settings!!
        )
    });

    operator fun get(config: AppSettingsState): TranslationRequest {
        return fn.apply(config)
    }
}